src/sentry/static/sentry/app/views/substances/uploadSubstancesButton.jsx

```diff
@@ -0,0 +1,187 @@
+            this.onToggle();
+            this.setState({
+              state: FormState.READY,
+              errors: {},
+            });
+          },
+          error: err => {
+            let errors = err.responseJSON || true;
+            errors = errors.detail || true;
+            this.setState({
+              state: FormState.ERROR,
+              errors,
+            });
+          },
+          complete: () => {
+            IndicatorStore.remove(loadingIndicator);
+          },
+        });
+      }.bind(this);
+
+      reader.onError = function() {
+        this.setState({
+          state: FormState.ERROR,
+        });
+      };
+    });
+  }
+
+  render() {
+    const isSaving = this.state.state === FormState.SAVING;
+    return (
+      <React.Fragment>
+        <a
+          title={this.props.tooltip || this.props.buttonTitle}
+          className="btn btn-sm btn-default"
+          disabled={this.props.disabled}
+          onClick={this.onToggle}
+          style={this.props.style}
+        >
+          {this.props.children}
+        </a>
+        <Modal show={this.state.isModalOpen} animation={false} onHide={this.onToggle}>
+          <form onSubmit={this.onSubmit}>
+            <div className="modal-header">
+              <h4>{t('Upload file')}</h4>
+            </div>
+
+            {/* TODO: Make pretty! Look e.g. into avatar uploading which
+                      doesn't use the default file upload look */}
+            <div className="modal-body">
+              {this.state.state === FormState.ERROR && (
+                <div className="alert alert-error alert-block">
+                  {t(`Unable to upload the file. ${this.state.errors}`)}
+                </div>
+              )}
+              <input type="file" name="" id="" onChange={this.handleSelectedFile} />
+            </div>
+
+            <div className="modal-footer">
+              <button
+                type="button"
+                className="btn btn-default"
+                disabled={isSaving}
+                onClick={this.onToggle}
+              >
+                {t('Cancel')}
+              </button>
+              {/* todo: dev only */}
+              <button
+                type="button"
+                onClick={this.handleDownload}
+                className="btn btn-default"
+                disabled={isSaving}
+              >
+                {t('Download example')}
+              </button>
+              <button
+                type="button"
+                onClick={this.handleUpload}
+                className="btn btn-primary"
+                disabled={isSaving}
+              >
+                {t('Upload')}
+              </button>
+            </div>
+          </form>
+        </Modal>
+      </React.Fragment>
+    );
+  }
+}
+
+UploadSubstancesButton.propTypes = {
+  tooltip: PropTypes.string,
+  buttonTitle: PropTypes.string,
+  disabled: PropTypes.bool,
+};
+
+export default UploadSubstancesButton;
```

Comment:
I suppose since there is really no (major) state handling here there is no point of hooking it up to redux?

Reply:
I think so, at least for the moment. It might fit better with the whole solution to have everything in redux at a later state, but it felt like an overkill here.